integration/test_collection_multi_node.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+import weaviate
+from weaviate import Config
+from weaviate.collection.classes.config import (
+    CollectionConfig,
+    VectorizerFactory,
+    Property,
+    ReplicationConfigCreate,
+    ConsistencyLevel,
+    DataType,
+)
+from weaviate.collection.classes.data import DataObject
+from weaviate.collection.classes.grpc import MetadataQuery
+
+
+@pytest.fixture(scope="module")
+def client():
+    client = weaviate.Client(
+        "http://localhost:8087", additional_config=Config(grpc_port_experimental=50058)
+    )
+    client.schema.delete_all()
+    yield client
+    client.schema.delete_all()
+
+
+@pytest.mark.parametrize(
+    "level", [ConsistencyLevel.ONE, ConsistencyLevel.ALL, ConsistencyLevel.QUORUM]
+)
+def test_consistency_on_multinode(client: weaviate.Client, level: ConsistencyLevel):
+    name = "TestConsistency"
+    client.collection.delete(name)
+    collection = client.collection.create(
+        CollectionConfig(
+            name=name,
+            vectorizer_config=VectorizerFactory.none(),
+            properties=[
+                Property(name="name", data_type=DataType.TEXT),
+            ],
+            replication_config=ReplicationConfigCreate(factor=2),
+        )
+    ).with_consistency_level(level)
+
+    collection.data.insert({"name": "first"})
+    ret = collection.data.insert_many(
+        [DataObject(properties={"name": "second"}), DataObject(properties={"name": "third"})]
+    )
+    assert not ret.has_errors
+
+    objects = collection.query.get_flat(return_metadata=MetadataQuery(is_consistent=True))
+    for obj in objects:
+        assert obj.metadata.is_consistent
```

Comment:
For my understanding, if you were to insert an object on one consistency level and then query on another would you expect the inserted object to be absent from the query?

Would it also be useful to add tests of `ConsistencyLevel` for the `collection.data` methods similar to this one?

Reply:
> For my understanding, if you were to insert an object on one consistency level and then query on another would you expect the inserted object to be absent from the query?

I'm not really sure 😅 AFAIK this is a per-query setting: So if you add with `One` vs `All`, All will take longer, because it waits for the reply of all nodes while One immediately gets back to you.

If you now query with `All` immediately after adding with `One`, you might get inconsistent results. If you query after a while it should be consistent

Reply:
Ah okay, that makes sense. I was getting confused as my question is actually more to do with the concept of multi-tenancy and was mixing up the two. I guess you can't really test, and perhaps shouldn't even, the eventual consistency on the client-side so there's no need for any tests to do so

Reply:
yeah, I see these tests more as "Did the right value arrive at the API" and not as "Did it work correctly in weaviate"